ui_test/src/rustc_stderr.rs

```diff
@@ -0,0 +1,144 @@
+use std::{
+    fmt::Write,
+    path::{Path, PathBuf},
+};
+
+use regex::Regex;
+
+#[derive(serde::Deserialize, Debug)]
+struct RustcMessage {
+    rendered: Option<String>,
+    spans: Vec<Span>,
```

Comment:
Can you move the type definitions together at the top of the file? I had to scroll quite a bit to figure out what a `Span` is.